### DISCUSSION TITLE
Usage with npx

### DISCUSSION THREAD

**Discussion by gimre** [dotansimha/graphql-code-generator#9140] [Q&A]
Hey, trying to use `grapql-code-generator` with npx but not getting anywhere :(

### Context
Maintaining a npm workspaces repository with multiple packages and looking to migrate from apollo codegen to `grapql-code-generator`. Previous setup didn't include apollo cli in any of the packages but expected developers to have it installed on their machine. Codegen would be defined and run at the package level like this ~
```package.json
"scripts": {
   "codegen": "apollo codegen [...]"
}
```

```
npm run codegen -w <worskpace_name>
```

### Problem
While configuring `grapql-code-generator` I installed `@graphql-codegen/cli` and '@graphql-codegen/typescript` inside one of the packages that already had `graphql` as a dependency and that ended up working fine.

Trying to generalize for all packages, i tried switching to
```package.json
"scripts": {
   "codegen": "npx -p @graphl-codegen/cli grapql-code-generator"
}
```
but this results the unresolved peerDependency `graphql`. Did not find a way to force npx to install graphql aswell before running the cli.

Tried installing `graphql`, `@graphl-codegen/cli` and `@graphql-codegen/typescript` globally and that will resolve the `graphql` dependency issue but will break with 
```
    ✖ Invalid Custom Plugin "typescript"
      Plugin typescript does not export a valid JS object with "plugin" function.
      Make sure your custom plugin is written in the following form:
      module.exports = {
      plugin: (schema, documents, config) => {
      return 'my-custom-plugin-content';
      },
      };
```

I assume this is due to a resolution issue, since `codegen.ts` is inside a workspace package and everything else is in npm cache.

All this happens if i try to install packages at the root level aswell.

### So...
Is there any way to either use the cli via npx or at least have it installed at the root level once instead of inside every workspace package?

**Comment by n1ru4l** [comment]
Hey @gimre, can you share a reproduction respository of your issue? Adding codegen and graphql as a dev dependency to the root workspace should just work fine
